docs/block-processing/slog.md

```diff
@@ -0,0 +1,68 @@
+Slog is a part of block processing that does things which don't fit in any other part (other parts being `Toil` in `txp`, `Poll` in `update`, `Toss` in `ssc`, and `lrc`).
+
+## What it does
+
+Slog is responsible for the following (see explanations below):
+
+* Pure verification
+* Ensuring that leaders from genesis blocks match leaders from LRC
+* State maintenance:
+    * Current tip
+    * `lastBlkSlots`, a list of slots of the last `blkSecurityParam` blocks
+    * Max seen difficulty
+    * Forward links
+    * `inMainChain` flags
+* Calling callbacks (`BListener`)
+
+Pure verification is *TODO DESCRIBE*
+
+Genesis blocks do not contain any information that cannot be derived from elsewhere, but we still store them as physical entities for convenience. Here we want to ensure that the list of leaders that is stored within genesis blocks is the same as the one computed by LRC based on the previous epoch.
+
+The list of slots of the last `blkSecurityParam` blocks (the last of which must be the tip) is an optimization for computing chain density.
+
+We maintain maximum seen difficulty to ensure that we never rollback deeper than `blkSecurityParam` from it.
```

Comment:
"Maximum seen difficulty" = "the highest difficulty ever seen in GState", not to be confused with "seen somewhere on the internet", right?

Reply:
Corrected.